Makefile.work

```diff
@@ -415,87 +498,75 @@ SONIC_BUILD_INSTRUCTION :=  make \
 
 .PHONY: sonic-slave-build sonic-slave-bash init reset
 
+
+ifeq ($(filter clean,$(MAKECMDGOALS)),)
+COLLECT_BUILD_VERSION = { DBGOPT='$(DBGOPT)' scripts/collect_build_version_files.sh $$?; }
+endif
+
+ifdef SOURCE_FOLDER
+	DOCKER_RUN += -v $(SOURCE_FOLDER):/var/$(USER)/src
+endif
+
+ifeq "$(KEEP_SLAVE_ON)" "yes"
+SLAVE_SHELL={ /bin/bash; }
+endif
+
 .DEFAULT_GOAL :=  all
 
-%::
+%:: | sonic-build-hooks
```

Comment:
Why?

Reply:
There is no dependency check for the sonic-build-hooks target, so we are checking this target always.